[ews-build.webkit.org] Use new-style test in TestWithFailureCount

JonWBedard opened the pull request:
#### 26498f2bd9a1111d2e94db122b993faddf1aa1b7
<pre>
[ews-build.webkit.org] Use new-style test in TestWithFailureCount
<a href="https://bugs.webkit.org/show_bug.cgi?id=249729">https://bugs.webkit.org/show_bug.cgi?id=249729</a>
rdar://103604782

Reviewed by Dewei Zhu.

* Tools/CISupport/ews-build/steps.py:
(TestWithFailureCount.run): Convert start to new-style &apos;run&apos; commands.
(TestWithFailureCount.countFailures): Accept returncode instead of command result.
(CheckStyle.countFailures): Accept returncode instead of command result.
(customBuildFlag): Renamed from appendCustomBuildFlags, detatch from step type.
(CompileWebKit.start): Replace appendCustomBuildFlags with customBuildFlag.
(RunJavaScriptCoreTests.start): Ditto.
(RunWebKitTests.setLayoutTestCommand): Ditto.
(RunAPITests.run): Convert start to new-style &apos;run&apos; command.
(RunAPITests.countFailures): Accept returncode instead of command result.
(RunAPITests.doOnFailure): Set of specific actions to take upon failure.
(ReRunAPITests.doOnFailure): Ditto.
(RunAPITestsWithoutChange):
(TestWithFailureCount.start): Deleted.
(TestWithFailureCount.commandComplete): Deleted.
(TestWithFailureCount.evaluateCommand): Deleted.
(appendCustomBuildFlags): Renamed customBuildFlag.
(RunAPITests.start): Deleted.
(RunAPITests.evaluateCommand): Deleted.
(ReRunAPITests.evaluateCommand): Deleted.
(RunAPITestsWithoutChange.evaluateCommand): Deleted.

Canonical link: <a href="https://commits.webkit.org/258497@main">https://commits.webkit.org/258497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36126a67b2c3f446d716c94b84d9442bf461c0bd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/101257 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10417 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/34319 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110538 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/170819 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/105242 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/11355 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/1281 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93657 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108370 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/107039 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8630 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91871 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/35161 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90532 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/23277 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/78159 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/4050 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/24791 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/4093 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/1219 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/100174 "Passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/10199 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/44270 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5850 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3074 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

dewei-zhu commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4367,70 +4363,65 @@ class RunAPITests(TestWithFailureCount):
-            command = ['python3', 'Tools/Scripts/run-gtk-tests',
+            self.command = ['python3', 'Tools/Scripts/run-gtk-tests',
                        '--{0}'.format(self.getProperty('configuration')),
                        '--json-output={0}'.format(self.jsonFileName)]
```
Those indentation also need to be updated?
Also if this is python3 only, we can do 
```
...
'--json-output={0}'.format(self.jsonFileName), *customBuildFlag(platform, self.getProperty('fullPlatform'))]
```

dewei-zhu reviewed (approved):
r=me

JonWBedard committed:
[ews-build.webkit.org] Use new-style test in TestWithFailureCount
https://bugs.webkit.org/show_bug.cgi?id=249729
rdar://103604782

Reviewed by Dewei Zhu.

* Tools/CISupport/ews-build/steps.py:
(TestWithFailureCount.run): Convert start to new-style 'run' commands.
(TestWithFailureCount.countFailures): Accept returncode instead of command result.
(CheckStyle.countFailures): Accept returncode instead of command result.
(customBuildFlag): Renamed from appendCustomBuildFlags, detatch from step type.
(CompileWebKit.start): Replace appendCustomBuildFlags with customBuildFlag.
(RunJavaScriptCoreTests.start): Ditto.
(RunWebKitTests.setLayoutTestCommand): Ditto.
(RunAPITests.run): Convert start to new-style 'run' command.
(RunAPITests.countFailures): Accept returncode instead of command result.
(RunAPITests.doOnFailure): Set of specific actions to take upon failure.
(ReRunAPITests.doOnFailure): Ditto.
(RunAPITestsWithoutChange):
(TestWithFailureCount.start): Deleted.
(TestWithFailureCount.commandComplete): Deleted.
(TestWithFailureCount.evaluateCommand): Deleted.
(appendCustomBuildFlags): Renamed customBuildFlag.
(RunAPITests.start): Deleted.
(RunAPITests.evaluateCommand): Deleted.
(ReRunAPITests.evaluateCommand): Deleted.
(RunAPITestsWithoutChange.evaluateCommand): Deleted.

Canonical link: https://commits.webkit.org/258497@main
